### PR TITLE
Rachana_Assignment_17_45min_Done

### PR DESCRIPTION
**Pull request by RachanaGJun20**
Asmnt17
asmnt17_2

### CODE REVIEW

File: src/rachanaGJun20/String/StringCharCounting.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		int uppercasecount = 0;
+		int letterscount = 0;
+		int specialcharcount = 0;
+
+		for (int i = 0; i < inputstr.length(); i++) {
+
+			if (Character.isLetter(inputstr.charAt(i))) {
+				letterscount++;
+				if (Character.isLowerCase(inputstr.charAt(i)))
+					lowercasecount++;
```
Comment (KrishnaKTechnocredits) [inline src/rachanaGJun20/String/StringCharCounting.java]: Here nested if not required for this letter logic. Think here. Also this variable letterscount  is not required.

File: src/rachanaGJun20/String/StringIndexOfChar.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+public class StringIndexOfChar {
+
+	void printFirstAndLastIndexOfChar(String str) { // method will print first and last index of a char if char
+													// occurrence is more than one
+
+		for (int i = 0; i < str.length(); i++) {
+			int charcount = 0;
+			if (i == str.indexOf(str.charAt(i))) {
+				for (int j = 0; j < str.length(); j++) {
+					if (str.charAt(i) == str.charAt(j))
```
Comment (KrishnaKTechnocredits) [inline src/rachanaGJun20/String/StringIndexOfChar.java]: You have increased the complexity in code, Using one for loop you can achieve the output.

File: src/rachanaGJun20/String/StringCharCounting.java (hunk 1 above)
Comment (RachanaGJun20) [reply src/rachanaGJun20/String/StringCharCounting.java]: ok got it got it!

File: src/rachanaGJun20/String/StringCharCounting.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+		int digitcount = 0;
+		int lowercasecount = 0;
+		int uppercasecount = 0;
+		int specialcharcount = 0;
+
+		for (int i = 0; i < inputstr.length(); i++) {
+
+			if (Character.isLowerCase(inputstr.charAt(i)))
+				lowercasecount++;
+			else if (Character.isUpperCase(inputstr.charAt(i)))
```
Comment (KrishnaKTechnocredits) [inline src/rachanaGJun20/String/StringCharCounting.java:26]: Perfect logic.

File: src/rachanaGJun20/String/StringCharCounting.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+				uppercasecount++;
+
+			else if (Character.isDigit(inputstr.charAt(i)))
+				digitcount++;
+			else
+				specialcharcount++;
+		}
+		System.out.println("Total Digits: " + digitcount + "\nTotal letters: " + (lowercasecount+uppercasecount) + "\nTotal Uppercase: "
```
Comment (KrishnaKTechnocredits) [inline src/rachanaGJun20/String/StringCharCounting.java:34]: Good job.

**Review by KrishnaKTechnocredits — APPROVED** [review]

**Pull request merged**